src/test/shell/integration/aquery_test.sh

```diff
@@ -117,11 +117,37 @@ EOF
   fi
 
   assert_contains "echo unused" output
-  bazel aquery --noinclude_commandline "//$pkg:bar" > output \
+  bazel aquery --output=text --noinclude_commandline "//$pkg:bar" > output \
     2> "$TEST_log" || fail "Expected success"
   assert_not_contains "echo unused" output
 }
 
+function test_aquery_textproto() {
+  local pkg="${FUNCNAME[0]}"
+  mkdir -p "$pkg" || fail "mkdir -p $pkg"
+  cat > "$pkg/BUILD" <<'EOF'
+genrule(
+    name = "bar",
+    srcs = ["dummy.txt"],
+    outs = ["bar_out.txt"],
+    cmd = "echo unused > $(OUTS)",
+)
+EOF
+  echo "hello aquery" > "$pkg/in.txt"
+
+  bazel aquery --output=textproto "//$pkg:bar" > output 2> "$TEST_log" \
+    || fail "Expected success"
+  cat output >> "$TEST_log"
+  assert_contains "exec_path: \"$pkg/dummy.txt\"" output || fail "Wrong output"
```

Comment:
assert_contains already fails with a good error message, see:
https://source.bazel.build/bazel/+/893e719801c58df3025474a6c5bc27ee3bcb3b4f:src/test/shell/unittest.bash;l=571

So you can remove all the || fail from the following lines

Reply:
I added those on purpose. assert_contains fails the test but continues executing, which leaves a lot of test output and makes it hard to find the actual failure cause. Failing immediately when assert_contains has an error makes it much easier to track the actual issue.

Reply:
huh, I'm surprised. Is this behavior expected? assert_contains also calls fail which calls exit so I assumed it would stop executing this test.